controller_manager:
  ros__parameters:
    update_rate: 500

joint_state_broadcaster:
  ros__parameters:
    type: joint_state_broadcaster/JointStateBroadcaster
    state_publish_rate: 200.0

ur5e_controller:
  ros__parameters:
    type: joint_trajectory_controller/JointTrajectoryController
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

    # publicação / action
    state_publish_rate: 50.0

    # usa sempre float para estas
    action_monitor_rate: 20.0

    # gains P/D
    gains:
      shoulder_pan_joint:
        p: 150.0
        d:   2.0
      shoulder_lift_joint:
        p: 150.0
        d:   2.0
      elbow_joint:
        p: 100.0
        d:   1.0
      wrist_1_joint:
        p:  60.0
        d:   0.5
      wrist_2_joint:
        p:  60.0
        d:   0.5
      wrist_3_joint:
        p:  40.0
        d:   0.3

    interpolation_method: splines
    allow_nonzero_velocity_at_trajectory_end: true

    goal_time_allowance: 0.5

    # tolerâncias
    constraints:
      goal_time:                  0.1
      stopped_velocity_tolerance: 0.02
      aborted_goal_tolerance:     0.05

